### PR TITLE
[21309] Remove 2.1.x branch from nightly jobs

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

The 2.1.x branch is associated to Fast DDS 2.13.x, wich is now EOL.
This PR removes the nightly jobs for that version

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: CI pass and failing tests are unrelated with the changes.

### CODE REVIEW

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
Nightly manual runs:

* [macOS](https://github.com/eProsima/Fast-CDR/actions/runs/9904059292)
* [Ubuntu](https://github.com/eProsima/Fast-CDR/actions/runs/9904061590)
* [Windows](https://github.com/eProsima/Fast-CDR/actions/runs/9904062493)

File: .github/workflows/nightly-mac-ci.yml
Comment (EduPonz) [inline .github/workflows/nightly-mac-ci.yml:1]: `nightly-mac-ci-1_0_x` should run on `macos-13` image, not `mac-13`

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by JesusPoderoso** [comment]
Nightly manual runs:

* [macOS](https://github.com/eProsima/Fast-CDR/actions/runs/9904317587)
* [Ubuntu](https://github.com/eProsima/Fast-CDR/actions/runs/9904318452)
* [Windows](https://github.com/eProsima/Fast-CDR/actions/runs/9904319445)

**Review by EduPonz — APPROVED** [review]
LGTM with green manual nightly CI

**Pull request merged**